frontend/src/Hooks/useFetchData.ts

```diff
@@ -2,19 +2,24 @@ import {useEffect, useState} from "react";
 import {Bike} from "../model/Bike";
 import axios from "axios";
 import useAuth from "./useAuth";
+import {ServiceOrder} from "../model/ServiceOrder";
 
-export default function useBikes(){
+export default function useFetchData(){
     const [bikes, setBikes] = useState<Bike[]>([])
+    const [orders, setOrders] = useState<ServiceOrder[]>([])
     const user = useAuth(false)
-    useEffect(() =>fetchBikes(), [user])
+    useEffect(() =>fetchData(), [user])
```

Comment:
Don't you have to include `fetchData` in your dependency array? Did you check for a warning like that? If so, it could help to use `useCallback`. You still would need to include it, but at least the method stays the same. (I haven't used it yet, so I cannot tell you much about it)